### PR TITLE
LEAF 2034 – Add file tracking 

### PR DESCRIPTION
**Pull request by jrodcode**
In order to test this, there has to be an update of the db, which will be added after this commit.

### CODE REVIEW

**Review by maxrichardjr — APPROVED** [review]

**Review by maxrichardjr — APPROVED** [review]

File: LEAF_Request_Portal/sources/TemplateFileHistory.php [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+                WHERE file_name = :templateFile
+                ORDER BY `file_created` DESC';
+
+        return $this->db->prepared_query($sql, $vars);
+    }
+
+
+    public function getTemplateFileHistory($templateFile)
+    {
+        error_log(print_r($templateFile, true));
```
Comment (raphaelfretter) [inline LEAF_Request_Portal/sources/TemplateFileHistory.php:84]: Why are we printing to error log here?

**Review by raphaelfretter — CHANGES_REQUESTED** [review]

File: LEAF_Request_Portal/sources/TemplateFileHistory.php (hunk 1 above)
Comment (jrodcode) [reply LEAF_Request_Portal/sources/TemplateFileHistory.php:84]: I have updated that section

**Review by maxrichardjr — APPROVED** [review]

**Review by maxrichardjr — APPROVED** [review]

**Review by maxrichardjr — APPROVED** [review]

**Review by maxrichardjr — APPROVED** [review]

**Review by maxrichardjr — APPROVED** [review]

**Review by maxrichardjr — APPROVED** [review]

**Review by maxrichardjr — APPROVED** [review]

File: LEAF_Request_Portal/sources/EmailTemplate.php [hunk 2]
```diff
@@ -168,16 +180,15 @@ public function getHistory($filterByName)
 
     public function setEmailTemplate($template)
     {
-        if (!$this->login->checkGroup(1))
-        {
+        error_log(print_r('setEmailTemplate - Template Name: '. $template, true));
```
Comment (shaneodd) [inline LEAF_Request_Portal/sources/EmailTemplate.php:183]: Why are you dumping to the error log here?

File: LEAF_Request_Portal/sources/EmailTemplate.php [hunk 3]
```diff
@@ -24,7 +24,8 @@ public function __construct($db, $login)
         $this->dataActionLogger = new \Leaf\DataActionLogger($db, $login);
     }
 
-    public function isEmailTemplateValid($template, $list) {
+    public function isEmailTemplateValid($template, $list)
```
Comment (shaneodd) [inline LEAF_Request_Portal/sources/EmailTemplate.php:27]: This file could use a look at typing.

File: LEAF_Request_Portal/sources/EmailTemplate.php [hunk 4]
```diff
@@ -230,34 +249,115 @@ public function setEmailTemplate($template)
...
+        if (!$validTemplate) {
+            return 'Invalid email template';
+        }
+
+        $time = date("Y-m-d h:i:s");
+        $templateID = uniqid();
+
+        // Update email body file
+        if (!empty($_POST['file'])) {
+            error_log(print_r($_POST['file'], true));
```
Comment (shaneodd) [inline LEAF_Request_Portal/sources/EmailTemplate.php:274]: error logging?

File: LEAF_Request_Portal/sources/TemplateFileHistory.php [hunk 5]
```diff
@@ -0,0 +1,145 @@
+<?php
```
Comment (shaneodd) [inline LEAF_Request_Portal/sources/TemplateFileHistory.php:1]: This entire file should be gone through with the psr 12 coding standards

File: LEAF_Request_Portal/templates_history/email_templates/642b1e013cbd6_LEAF_main_email_template.tpl [hunk 6]
```diff
@@ -0,0 +1,7 @@
+<html>
```
Comment (shaneodd) [inline LEAF_Request_Portal/templates_history/email_templates/642b1e013cbd6_LEAF_main_email_template.tpl:1]: Is this a test file?

**Review by shaneodd — CHANGES_REQUESTED** [review]
Let me know if you have questions. Here is the guide I have been using for psr-12 https://www.php-fig.org/psr/psr-12/

**Review by maxrichardjr — APPROVED** [review]

**Review by maxrichardjr — APPROVED** [review]

**Review by raphaelfretter — APPROVED** [review]

**Review by jampaul3 — APPROVED** [review]

File: LEAF_Request_Portal/sources/TemplateFileHistory.php (hunk 5 above)
Comment (jrodcode) [reply LEAF_Request_Portal/sources/TemplateFileHistory.php:1]: I have made the changes

File: LEAF_Request_Portal/sources/EmailTemplate.php (hunk 2 above)
Comment (jrodcode) [reply LEAF_Request_Portal/sources/EmailTemplate.php]: Has been removed

File: LEAF_Request_Portal/sources/EmailTemplate.php (hunk 4 above)
Comment (jrodcode) [reply LEAF_Request_Portal/sources/EmailTemplate.php]: Has been removed

**Review by maxrichardjr — APPROVED** [review]

File: LEAF_Request_Portal/sources/EmailTemplate.php (hunk 3 above)
Comment (jrodcode) [reply LEAF_Request_Portal/sources/EmailTemplate.php]: I have updated the file.

**Review by Pelentan — CHANGES_REQUESTED** [review]
This needs major cleanup.  It has commits from other branches.

**Review by maxrichardjr — APPROVED** [review]

**Pull request closed without merging**